### PR TITLE
Homework4

### PR DESCRIPTION
**Pull request by m-kop**
- InvalidGear exception
- Unit tests
- accelerate() interface fix
- unique_ptr's for engines
- member fields made private to avoid e.g. changing petrol engine capacity

### CODE REVIEW

File: Car.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    std::cout << __FUNCTION__ << std::endl;
+}
+void Car::brake()
+{
+    std::cout << __FUNCTION__ << std::endl;
+}
+void Car::accelerate(int speed)
+{
+    if (speed < 0) {
+        return;
```
Comment (WBigaj) [inline Car.cpp:20]: Isn't better to throw some exception here?

**Review by WBigaj — COMMENTED** [review]
I have checked your code and have one point.

File: Car.cpp (hunk 1 above)
Comment (MaciejSalamonski) [reply Car.cpp:20]: +1

**Review by MaciejSalamonski — APPROVED** [review]
Hi nice job! Keep it up!

**Comment by ziobron** [comment]
4 XP za wyjątek InvalidGear w Cars (3521)
4 * 2 XP za poprawkę interfejsu w Cars (3522)
10 XP za testy w Cars (3523)
3 * 2 XP za punktualność (1119)
{ "requirement_ids": [3521, 3523, 3522, 3522, 3522, 3522, 1119, 1119, 1119] }

**Pull request closed without merging**